web/hooks/use-group-supabase.ts

```diff
@@ -0,0 +1,202 @@
+  const user = useUser()
+  const isManifoldAdmin = useAdmin()
+  useEffect(() => {
+    setTranslatedMemberRole(groupId, isManifoldAdmin, setUserRole, user)
+  }, [user])
+  useEffect(() => {
+    const channel = db.channel('user-group-role-realtime')
+    channel.on(
+      'postgres_changes',
+      {
+        event: '*',
+        schema: 'public',
+        table: 'group_members',
+        filter: `group_id=eq.${groupId}`,
+      },
+      (payload) => {
+        if (payload.new.member_id === user?.id) {
+          setTranslatedMemberRole(groupId, isManifoldAdmin, setUserRole, user)
+        }
+      }
+    )
+    channel.subscribe(async (status) => {})
+    return () => {
+      db.removeChannel(channel)
+    }
+  }, [db, user])
+  return userRole
+}
+
+export function useRealtimeGroupMembers(
+  groupId: string,
+  hitBottom: boolean,
+  numMembers: number | undefined
+) {
+  const [admins, setAdmins] = useState<JSONContent[] | undefined>(undefined)
+  const [moderators, setModerators] = useState<JSONContent[] | undefined>(
+    undefined
+  )
+  const [members, setMembers] = useState<JSONContent[] | undefined>(undefined)
+  const [loadMore, setLoadMore] = useState<boolean>(false)
+  const [offsetPage, setOffsetPage] = useState<number>(0)
+
+  function loadMoreMembers() {
+    setLoadMore(true)
+    getGroupMembers(groupId, offsetPage + 1)
+      .then((result) => {
+        if (members) {
+          const prevMembers = members
+          setMembers([...prevMembers, ...result.data])
+        } else {
+          setMembers(result.data)
+        }
+        setOffsetPage((offsetPage) => offsetPage + 1)
+      })
+      .catch((e) => console.log(e))
+      .finally(() => setLoadMore(false))
+  }
+  function fetchGroupMembers() {
+    getGroupOfRole(groupId, 'admin')
+      .then((result) => {
+        const admins = result.data
+        setAdmins(admins)
+      })
+      .catch((e) => console.log(e))
+
+    getGroupOfRole(groupId, 'moderator')
+      .then((result) => {
+        const moderators = result.data
+        setModerators(moderators)
+      })
+      .catch((e) => console.log(e))
+
+    getGroupMembers(groupId, offsetPage, 0)
+      .then((result) => {
+        const members = result.data
+        setMembers(members)
+      })
+      .catch((e) => console.log(e))
+  }
+
+  useEffect(() => {
+    fetchGroupMembers()
+  }, [])
+
+  useEffect(() => {
+    if (hitBottom && !loadMore && numMembers && numMembers > MEMBER_LOAD_NUM) {
+      loadMoreMembers()
+    }
+  }, [hitBottom])
+
+  useEffect(() => {
+    const channel = db.channel('group-members-realtime')
+    channel.on(
+      'postgres_changes',
+      {
+        event: '*',
+        schema: 'public',
+        table: 'group_members',
+        filter: `group_id=eq.${groupId}`,
+      },
+      (payload) => {
+        fetchGroupMembers()
+      }
+    )
+    channel.subscribe(async (status) => {})
+    return () => {
+      db.removeChannel(channel)
+    }
```

Comment:
Nice! Cool to see this working. Seems a little awkward to fetch again when there's a change (payload might contain what changed?), but I guess this is simpler.